### DISCUSSION TITLE
Directives run again, even outside x-data

### DISCUSSION THREAD

**Discussion by BregtDeLange** [alpinejs/alpine#2991] [1. Help] [answered]
I have a list of cards, rendered with x-for in Alpine. On this list is a listener for lightGallery. When clicking on a card, lightGallery will open up the lightbox, show the card's image and a bit of HTML. This HTML is a copy of a subset of original card's HTML. This is possible in lightGallery with the `data-sub-html` attribute on the container. lightGallery copies the children of the element passed to `data-sub-html` to the caption part of the lightbox for that image.

Here it breaks. Whereas in v2, this seemed to worked fine, and nothing happens, which is what I want, with the upgrade to v3 of Alpine, it seems Alpine wants to run the code for this caption again, even though it is outside the x-data scope. Because lightGallery copies the HTML to a div at the end of the <body>.

Error:
`Uncaught ReferenceError: x is not defined`

I tried adding x-ignore to the captions container, but this does not work.

I created **a simplified Codepen**, without lightGallery, for you to see what is happening. When I copy a selection of the HTML of a list item out of the x-data, this will result in an error (Uncaught ReferenceError: color is not defined) 
https://codepen.io/Bregt/pen/ZErwmNa?editors=1010

It did not return errors with v2. I do not understand why Alpine reinterpretes the directives outside an x-data scope. lightGallery is not the cause of the issue, as the Codepen above is an example without it.

AlpineJS version: ~3.10.2
lightGallery version: 2.1.8

**Comment by Tim-Wils** [comment, marked as answer]
Hi! 
Try using Alpinejs.mutateDom() like so ( https://codepen.io/onstuimig/pen/RwQdqay?editors=1010 ) to pause Alpine from listening to mutations on the dom. That way you can safely copy your elements.

**Reply by BregtDeLange** [reply]
Thanks Tim. I hadn't read about that function. Is this documented somewhere, or is it supposed or preferred not to be used?
